tensorflow/python/compiler/tensorrt/test/tf_trt_integration_test_base.py

```diff
@@ -157,8 +157,27 @@ def setUp(self):
     super(TfTrtIntegrationTestBase, self).setUp()
     warnings.simplefilter("always")
 
-  def BuildParams(self, graph_fn, dtype, input_shapes, output_shapes):
-    """Build test parameters when not considering dynamic shapes."""
+  def _GetTensorSpec(self, shape, mask, dtype, name):
+    # Unset shape where mask[i] == None
+    assert len(shape) == len(mask)
+    new_shape = [None if m is False else s for s, m in zip(shape, mask)]
+    return tensor_spec.TensorSpec(new_shape, dtype, name)
+
+  def BuildParams(self, graph_fn, dtype, input_shapes, output_shapes,
```

Comment:
The special handling for implicit batch mode makes BuildParams non-trivial to understand. Can we have two routines instead:
         BuildParamsForImplicitBatchMode
         BuildParamsForMasks
The second one is similar to the current BuildParams with the exception that it doesn't allow default mask values. The first one constructs the masks for implicit batch mode and calls the second one.

Reply:
Yes, I was contemplating about such change myself too. I have kept the implicit batch version as BuildParams, because it is used in most of the tests.